### PR TITLE
[EXPERIMENTAL] Provide addon-defined template types for Glint

### PR DESCRIPTION
**Pull request by kpfefferle**
This PR adds Glint template types for Ember Table to be published with the addon package. It enables consumers to define the shape of their table rows and columns and then enforces that the arguments passed in to Ember Table matches those types. All publicly documented arguments are also enforced to be the appropriate type. Yielded values (rows, columns, cell values) are yielded with respective types as well (with cell values yielded as a union of possible types).

These Glint types are isolated to the `/types/` directory in the repo, so they should not affect any of the functional addon code.

# Documentation - as added to the `README`

## EXPERIMENTAL: Using Ember Table with Glint

Ember Table provides **experimental** Glint types defined in the `/types/` directory.
These types may change at any time and are **NOT** covered by Ember Table's semantic versioning.
They are intended to support standard documented usage of Ember Table and do not attempt to type the internals of the Ember Table addon.
If you are using Ember Table in a more advanced way (such as extending Ember Table components), you will still need to define your own types for those use cases.

To use the provided Ember Table types for Glint:

1. Define a type interface for your row contents. If your columns contain additional custom attributes, you can type those as well. Ember Table provides default interfaces that can be extended for this purpose.
1. Extend the base Ember Table component passing in your row and (optional) column interfaces as generics.
1. Use this extended version of the Ember Table component in your template.

For example:

```ts
// my-table-component.ts
import type { EmberTableColumn, EmberTableRow } from 'ember-table';
import EmberTableComponent from 'ember-table/components/ember-table/component';

interface MyTableColumn extends EmberTableColumn {
  // Add any custom column attribute types here (optional)
}

interface MyTableRow extends EmberTableRow {
  // Add the attributes and types for your table rows here
}

class MyEmberTableComponent extends EmberTableComponent<MyTableRow, MyTableColumn> {}

export default class MyTableComponent extends Component<MyTableComponentSignature> {
  emberTableComponent = MyEmberTableComponent;
}
```

```hbs
{{! my-table-component.hbs }}
<this.emberTableComponent as |t|>
  {{! Use Ember Table as usual. Row and column arguments will be enforced to match the appropriate types. }}
  {{! Yielded items (rows, columns) will be typed according to the specified interfaces. }}
  {{! Cell values will be typed as a union of all defined row attribute types. }}
</this.emberTableComponent>
```

### CODE REVIEW

**Comment by kpfefferle** [comment]
Should migrate to the new recommendation for exporting registry entries: https://github.com/typed-ember/glint/issues/446

File: addon/components/ember-thead/component.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ export default Component.extend({
   sorts: defaultTo(() => []),
 
   /**
-    An optional sort. If not specified, defaults to <sortMultiple>, which
+    An optional sort. If not specified, defaults to `<sortMultiple>`, which
```
Comment (kpfefferle) [inline addon/components/ember-thead/component.js:90]: While typing documented arguments, I noticed that `<sortMultiple>` is not rendered in the API docs without the backticks

File: addon/components/ember-thead/component.js [hunk 2]
```diff
@@ -195,13 +195,6 @@ export default Component.extend({
   */
   containerWidthAdjustment: null,
 
-  /**
-    An action that is sent when sorts is updated
-    @argument onHeaderAction
-    @type Action?
-  */
-  onHeaderAction: null,
```
Comment (kpfefferle) [inline addon/components/ember-thead/component.js:203]: This argument appears to be completely unused - I couldn't find reference to it anywhere else in the codebase (and the comment is obviously inaccurately copied from `onUpdateSorts` below).

**Comment by kpfefferle** [comment]
Released as [5.0.1-0](https://www.npmjs.com/package/ember-table/v/5.0.1-0)

File: README.md [hunk 3]
```diff
@@ -157,6 +157,50 @@ import { setupForTest as setupEmberTableForTest } from 'ember-table/test-support
...
+```
+
+```hbs
+{{! my-table-component.hbs }}
+<this.emberTableComponent as |t|>
+  {{! Use Ember Table as usual. Row and column arguments will be enforced to match the appropriate types. }}
+  {{! Yielded items (rows, columns) will be typed according to the specified interfaces. }}
+  {{! Cell values will be typed as a union of all defined row attribute types. }}
+</this.emberTableComponent>
+```
```
Comment (mixonic) [inline README.md:202]: Thanks for these docs 🙇

**Review by mixonic — APPROVED** [review]

**Comment by mixonic** [comment]
Amazing.

**Comment by kpfefferle** [comment]
Released as [5.0.1](https://github.com/Addepar/ember-table/tree/v5.0.1)

**Pull request merged**